packer/builder/azure/arm/config.go

```diff
@@ -78,11 +105,24 @@ func (c *Config) toTemplateParameters() *TemplateParameters {
 		ImagePublisher:     &TemplateParameter{c.ImagePublisher},
 		ImageSku:           &TemplateParameter{c.ImageSku},
 		OSDiskName:         &TemplateParameter{c.tmpOSDiskName},
-		SshAuthorizedKey:   &TemplateParameter{c.sshAuthorizedKey},
 		StorageAccountName: &TemplateParameter{c.StorageAccount},
 		VMSize:             &TemplateParameter{c.VMSize},
 		VMName:             &TemplateParameter{c.tmpComputeName},
 	}
+
+	switch c.OSType {
+	case constants.GOOS_Linux:
```

Comment:
OK.

I would argue (weakly) we're duplicating existing constants, so why not use the existing ones as long as you *snap* to their semantics.